src/osp/tasks/top_utils.h

```diff
@@ -41,6 +41,11 @@
 namespace osp
 {
 
+/**
+ * @brief Reserves the first empty value found in topData after the index current
+ *
+ * @return The index of the reserved value, or if topData is full, the largest index in topData
```

Comment:
the index of the reserved value, or the index after the end if no slots are available after the indicated index.